src/anomalib/callbacks/metrics.py

```diff
@@ -98,6 +97,8 @@ def setup(
             else:
                 pl_module.pixel_metrics = create_metric_collection(pixel_metric_names, "pixel_")
             self._set_threshold(pl_module)
+            if hasattr(trainer.datamodule, "saturation_config"):
+                self._set_saturation_config(pl_module, trainer.datamodule.saturation_config)
```

Comment:
would it be possible to somehow handle this within the SPRO itself? This is metric specific, would be great if we could sort it out in the metric

Reply:
@samet-akcay do you mean loading the config directly in the metric? i think it would be possible as long as we introduce a new parameter in the `yaml`, so that we can load the config in the metric. I will do as follows if it is okay.
```
metrics:
  pixel:
    - AUROC
    - SPRO
  saturation_config: ./datasets/MVTec_LOCO/breakfast_box/defects_config.json
```

Reply:
@ashwinvaidya17, can you confirm whether we could use `class_path` and `init_args` for metrics above? The implementation would be quite straightforward

Reply:
@samet-akcay @ashwinvaidya17 In the latest implementation, I have handled the loading of the saturation config in the metric itself. You can see what I modified in this commit: https://github.com/openvinotoolkit/anomalib/pull/1686/commits/d9a233340061c70334f726869e4189d33e01c061.

I found there is an existing function, i.e., `metric_collection_from_dicts`, which allows us to use `class_path` and `init_args` by passing a `dict[str, dict[str, Any]]` to the function

https://github.com/openvinotoolkit/anomalib/blob/main/src/anomalib/metrics/__init__.py#L184-L186

However, the current CLI doesn't support the dict type. So I modified this part by adding the dict type to make it work:
https://github.com/openvinotoolkit/anomalib/blob/d9a233340061c70334f726869e4189d33e01c061/src/anomalib/cli/cli.py#L146-L151

I have tested this configuration in the YAML file, and the saturation_config path can be loaded on the metric side:

```yaml
metrics:
  pixel:
    SPRO:
      class_path: anomalib.metrics.SPRO
      init_args:
        saturation_config: ./datasets/MVTec_LOCO/pushpins/defects_config.json
```